txn_writer.go

```diff
@@ -252,6 +266,10 @@ func (sw *SafeWriter) Write(root string, sm gps.SourceManager) error {
 	defer os.RemoveAll(td)
 
 	if sw.Payload.HasManifest() {
+		if sw.Payload.Manifest.IsEmpty() {
+			err := modifyWithString(mpath, exampleToml)
+			return errors.Wrap(err, "failed to generate example text")
```

Comment:
This is unconditionally returning anytime we are writing the helptext, which means that anything else that should have been written (in this case, the lock file) isn't. That's why the travis build is failing (https://travis-ci.org/golang/dep/jobs/221895965#L190).

Also at this point in the code, the manifest should be written to the temp directory, instead of to the final location. The way the safe write works is that first we write to a temp folder, then we save copies of the original files, and then we move the updated files to the final location. If anything goes wrong, then we put back the original files.

I think something like this will get you past the error:

```go
if sw.Payload.HasManifest() {
	if sw.Payload.Manifest.IsEmpty() {
		err := modifyWithString(filepath.Join(td, ManifestName), exampleToml)
		if err != nil {
			return errors.Wrap(err, "failed to generate example text")
		}
	} else if err := writeFile(filepath.Join(td, ManifestName), sw.Payload.Manifest); err != nil {
		return errors.Wrap(err, "failed to write manifest file to temp dir")
	}
}
```